mysql 5.1 include files intentionally break user use of "bool", which is
probably a seriously bad idea; but for now we'll work around it here not
there.

Reported upstream at http://bugs.mysql.com/bug.php?id=42120


diff -Naur mysql-connector-odbc-5.1.5r1144.orig/driver/myutil.h mysql-connector-odbc-5.1.5r1144/driver/myutil.h
--- mysql-connector-odbc-5.1.5r1144.orig/driver/myutil.h	2008-04-14 21:03:13.000000000 -0400
+++ mysql-connector-odbc-5.1.5r1144/driver/myutil.h	2009-01-14 15:45:46.000000000 -0500
@@ -204,7 +204,7 @@
 				      SQLLEN *pcbValue);
 SQLRETURN SQL_API my_SQLExtendedFetch(SQLHSTMT hstmt, SQLUSMALLINT fFetchType,
 				      SQLROWOFFSET irow, SQLULEN *pcrow,
-				      SQLUSMALLINT FAR *rgfRowStatus, bool upd_status);
+				      SQLUSMALLINT FAR *rgfRowStatus, my_bool upd_status);
 SQLRETURN copy_stmt_error(STMT FAR *src, STMT FAR *dst);
 int unireg_to_c_datatype(MYSQL_FIELD *field);
 int default_c_type(int sql_data_type);
diff -Naur mysql-connector-odbc-5.1.5r1144.orig/driver/results.c mysql-connector-odbc-5.1.5r1144/driver/results.c
--- mysql-connector-odbc-5.1.5r1144.orig/driver/results.c	2008-06-06 15:37:04.000000000 -0400
+++ mysql-connector-odbc-5.1.5r1144/driver/results.c	2009-01-14 15:46:04.000000000 -0500
@@ -1184,7 +1184,7 @@
                                        SQLROWOFFSET         irow,
                                        SQLULEN             *pcrow,
                                        SQLUSMALLINT FAR    *rgfRowStatus,
-                                       bool                 upd_status )
+                                       my_bool              upd_status )
 {
     ulong rows_to_fetch;
     long cur_row, max_row;
